### PR TITLE
[POC] Generalise Psammead Brand to work with multiple services

### LINKED ISSUE #199: Psammead Brand should enable passing in different SVGs
**Is your feature request related to a problem? Please describe.**
Currently [`@bbc/psammead-brand`](https://github.com/BBC-News/psammead/tree/latest/packages/components/psammead-brand) is hard-coded to use the BBC News svg. It should be generalised to allow for the passing in of any other brand SVG.

Storybook link:
https://bbc-news.github.io/psammead/?selectedKind=Brand&selectedStory=default&full=0&addons=1&stories=1&panelRight=1&addonPanel=storybooks%2Fstorybook-addon-knobs

Note: we already pass in the `brandName` for offscreen text as a prop. https://github.com/BBC-News/psammead/blob/latest/packages/components/psammead-brand/src/index.jsx#L84

We note that currently, the height of the entire banner (at 80px) and the positioning of the hover/focus state just below the SVG are all working on the assumption that `SVG_HEIGHT_PX`  is fixed at 24px, which is the case for all News and World Service brand SVG assets. This issue is not to generalise the logic to allow for all heights of SVG. This is to only allow different brands with a 24px height SVG to be used. 

**Describe the solution you'd like**
`SVG_WIDTH_PX` is currently a fixed constant. This should be passed in as a prop (`svgWidth`) rather than being hard-coded. 
https://github.com/BBC-News/psammead/blob/latest/packages/components/psammead-brand/src/index.jsx#L29

SVG Brand Path can be passed in as a child element. 
https://github.com/BBC-News/psammead/blob/latest/packages/components/psammead-brand/src/index.jsx#L78-L82

So, example usage:
```jsx
<Brand svgWidth={167.95}>
  <g fill={`${C_WHITE}`} fillRule="evenodd">
    <path d="M84.32,0V24H58.82V0ZM79.26,16l-.41.25a11.91,11.91,0,0,1-6.07,1.85c-4.18,0-6.93-2.49-6.94-6.09s2.88-6.13,6.83-6.14a12,12,0,0,1,6,1.71l.4.22V4.6l-.17-.07A16.54,16.54,0,0,0,72.7,3.14a10.56,10.56,0,0,0-7.19,2.57,8.68,8.68,0,0,0-2.85,6.54A8.46,8.46,0,0,0,65,17.93a10,10,0,0,0,7.5,2.94h0a14,14,0,0,0,6.56-1.5l.15-.07Z M54.94,0V24H29.43V0ZM48.88,15.91a4.62,4.62,0,0,0-3.77-4.47,4.25,4.25,0,0,0,1.49-1.19,3.75,3.75,0,0,0,.72-2.34,4.17,4.17,0,0,0-1.39-3.13,6.33,6.33,0,0,0-4.36-1.41H36.12V20.65H42.5a7.18,7.18,0,0,0,4.9-1.53A4.27,4.27,0,0,0,48.88,15.91Z M45.7,15.57A2,2,0,0,1,45,17.16a4.22,4.22,0,0,1-2.85.79H39.22V13.23H42a4.89,4.89,0,0,1,2.82.68A1.93,1.93,0,0,1,45.7,15.57Z M43.32,9.92a2,2,0,0,0,.82-1.79,1.82,1.82,0,0,0-.58-1.43,3.33,3.33,0,0,0-2.26-.63H39.22v4.47h1.43A4.62,4.62,0,0,0,43.32,9.92Z M25.55,0V24H0V0ZM19.49,15.91a4.62,4.62,0,0,0-3.77-4.47,4.24,4.24,0,0,0,1.49-1.19,3.75,3.75,0,0,0,.72-2.34,4.18,4.18,0,0,0-1.39-3.13,6.33,6.33,0,0,0-4.36-1.41H6.74V20.65h6.38A7.18,7.18,0,0,0,18,19.12,4.27,4.27,0,0,0,19.49,15.91Z M13.93,9.92a2,2,0,0,0,.82-1.79,1.82,1.82,0,0,0-.58-1.43,3.33,3.33,0,0,0-2.26-.63H9.83v4.47h1.43A4.62,4.62,0,0,0,13.93,9.92Z M15.47,13.91a4.89,4.89,0,0,0-2.82-.68H9.83V18h2.94a4.23,4.23,0,0,0,2.85-.79,2,2,0,0,0,.69-1.59A1.93,1.93,0,0,0,15.47,13.91Z" />
    <path d="M111,3.36h2.36V20.7h-2.13L99.66,7.35V20.7H97.33V3.36h2L111,16.83Z" />
    <path d="m117.86 3.36h9.83v2.21h-7.35v5.29h7.1v2.22h-7.1v5.39h7.58v2.21h-10.06z M154.18,3.36h2.48l-7,17.41h-.55l-5.67-14.1-5.73,14.1h-.53l-7-17.41h2.5l4.78,12,4.81-12h2.35l4.83,12Z M163.38,13.43l-1.89-1.15A8.57,8.57,0,0,1,159,10.16a4,4,0,0,1-.75-2.41,4.26,4.26,0,0,1,1.42-3.33,5.31,5.31,0,0,1,3.69-1.28,7,7,0,0,1,4,1.22V7.17a5.74,5.74,0,0,0-4-1.8,3.34,3.34,0,0,0-2,.56,1.71,1.71,0,0,0-.78,1.44,2.22,2.22,0,0,0,.58,1.46,7.25,7.25,0,0,0,1.85,1.43l1.9,1.12Q168,13.28,168,16.21a4.42,4.42,0,0,1-1.4,3.39A5.11,5.11,0,0,1,163,20.9a7.63,7.63,0,0,1-4.68-1.58V16.17a5.84,5.84,0,0,0,4.65,2.55,2.93,2.93,0,0,0,1.94-.65,2,2,0,0,0,.78-1.63Q165.67,14.85,163.38,13.43Z" />
  </g>
</Brand>
```

I'm suggesting this rather than passing in the whole SVG so that the attributes required for accessibility are set in the Brand component https://github.com/BBC-News/psammead/blob/latest/packages/components/psammead-brand/src/index.jsx#L74-L76

This ticket does *not* address making the brand colour and font colour change via props, since this is not required for any of the News and World Service products. Will open up a separate issue for this, relevant for other products.

 
**Describe alternatives you've considered**
Passing in the whole SVG. This might seem easier, but it'll mean some of the attributes we require to make this component accessible won't be added to the svg. 

**Testing notes**
[Tester to complete]

Dev insight: There should be unit tests for two different Brands - e.g. News & one for a World Service site. There should be storybook stories for these two brands. https://bbc-news.github.io/psammead/?selectedKind=Brand&selectedStory=default&full=0&addons=1&stories=1

**Additional context**

- [x] Initially labelled with ["Refinement needed"](https://github.com/BBC-News/psammead/labels/Refinement%20Needed)

### PR DESCRIPTION
**Pull request by sareh**
Resolves #199

Generalising Psammead Brand

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: packages/components/psammead-brand/README.md [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
 
-## Usage 
-```
+## Usage
+
+| Prop    | Description     |
+| ------- | --------------- |
+| `brandName`  | Text equivalent of the Brand name, e.g. `BBC News`. Should be literal translation of the SVG text. Used for visually hidden text for users of Assistive Technologies. |
+| `children`   | React elements for Brand SVG paths, with all relevant SVG attributes as camel-cased (e.g. `stroke-width` as `strokeWidth`, [see React Docs](https://reactjs.org/docs/dom-elements.html#all-supported-html-attributes)) |
+| `svgWidth` | A number representing the width of the SVG, given that the height is fixed to 24px. Used for the viewbox width and width attribute of the SVG.  |
```
Comment (ChrisBAshton) [inline packages/components/psammead-brand/README.md:13]: I question the usefulness of the component if you have to pass an SVG to it 🤔 for me, I'd simplify the API to be:

```
name - as per brandName, but refactored name as we don't need to repeat "brand"
children - optional
```

If we recognise the `name` passed, we render the corresponding SVG (which is stored somewhere inside the component), which has its height & width defined too. So simple use-case would be:

```jsx
<Brand name="BBC News" />
```

For unrecognised brands, I'd propose this API:

```jsx
<Brand name="BBC Foo">
    <svg ...bla>
    </svg>
</Brand>
```

^ and in that case, we explicitly put in the documentation that any custom SVG has the width/height we require.

**Review by radiocontrolled — DISMISSED** [review]
I like how this approach, which treats the logo as a prop, prevents some bloat in psammead-brand and deputises responsibility for implementation considerations to the user of the component. Because of this, we might consider flagging any considerations for the component's implementers in the readme that we're not necessarily responsible for but that still could matter from their perspective. For example that this doesn't include a fallback or perhaps that's stating the obvious.

File: packages/components/psammead-brand/src/index.jsx [hunk 2]
```diff
@@ -60,37 +53,43 @@ const StyledSpan = styled.span`
...
-          </g>
-        </BrandSvg>
-        <VisuallyHiddenText>{brandName}</VisuallyHiddenText>
-      </StyledSpan>
-    </StyledLink>
-  </StyledWrapper>
-);
+  return (
+    <StyledWrapper>
+      <StyledLink href="https://www.bbc.co.uk/news">
```
Comment (dr3) [inline packages/components/psammead-brand/src/index.jsx]: The URL should be handled to be dynamic too

**Review by dr3 — CHANGES_REQUESTED** [review]

File: packages/components/psammead-brand/README.md (hunk 1 above)
Comment (sareh) [reply packages/components/psammead-brand/README.md:13]: My main reason for not including the SVG path in this brand component was that we wouldn't want to bloat this component with 40+ service SVGs.
Would we want to add them into this component?

**Comment by jamesbhobbs** [comment]
No need to keep this open as it's linked to on the issue.

**Pull request closed without merging**